### PR TITLE
📊 Update Statistical Review and energy dataset (for review)

### PR DESCRIPTION
**Pull request by pabloarosado**
## Summary

This PR updates the Statistical Review of World Energy (which is one of our core origins for energy data) as well as all steps that use it (including the `owid_energy` and `owid_co2` steps, that feed our `energy-data` and `co2-data` repos).

There are no fundamental changes to the code, but simply adaptations due to changes in the original data. Therefore, **no thorough review is required**.

## List of main changes:
* Created snapshots for new Statistical Review files and adapted old steps.
* Updated and adapted all energy steps.
* Updated the Global Carbon Budget dataset (to let it use the latest GDP and energy dependencies) and the `owid_co2` dataset (that feeds the `co2-data` repos).
* Moved `owid_co2` step  to the `external` channel.
* Archived `country_profile` step. It has never been used, and it introduces annoying dependencies.
* Archived unused energy and emissions steps.

## TODO (Pablo R):
* [x] Check if Saudi Arabia's wind generation [sudden increase](http://staging-site-update-statistical-review-subbranch/etl/wizard?page=chart-diff&chart_id=1412) in the last year is a data issue.

### CODE REVIEW

**Review by paarriagadap — APPROVED** [review]
Looks excellent!

I checked the changes and that all the steps ran properly. I guess etl-unit-tests failing is not a big deal and I know the staging error is due to [this problem](https://owid.slack.com/archives/C46U9LXRR/p1719244637246889), so you're good to go after checking what happens with Saudi Arabia :)

File: snapshots/energy_institute/2024-06-20/statistical_review_of_world_energy.xlsx.dvc [hunk 1]
```diff
@@ -2,17 +2,20 @@ meta:
...
+    url_download: https://www.energyinst.org/__data/assets/excel_doc/0020/1540550/EI-Stats-Review-All-Data.xlsx
     date_accessed: '2024-06-20'
-    date_published: '2023-06-26'
+    date_published: '2024-06-20'
     license:
-      name: © Energy Institute 2023
-      url: |-
-        https://web.archive.org/web/20230626200231/https://www.energyinst.org/__data/assets/pdf_file/0004/1055542/EI_Stat_Review_PDF_single-2.pdf
+      name: © Energy Institute 2024
+      url: https://www.energyinst.org/__data/assets/pdf_file/0006/1542714/EI_Stats_Review_2024_single_pages.pdf
```
[the remark this answers is not in the document]
Comment (pabloarosado) [reply snapshots/energy_institute/2024-06-20/statistical_review_of_world_energy.xlsx.dvc:17]: I haven't found it specifically talking about the Statistical Review. But I think that [this page](https://www.energyinst.org/terms) is probably good enough for license info.

File: etl/steps/data/garden/energy/2024-06-20/energy_mix.meta.yml [hunk 2]
```diff
@@ -40,7 +40,7 @@ definitions:
     The annual change in energy consumption by source is calculated as the difference with respect to the previous year.
   # Processing description for all per capita quantities.
   description_processing_per_capita: &description_processing_per_capita |-
-    Per capita figures are calculated by dividing by a population dataset that is built and maintained by Our World in Data, based on [different sources](https://ourworldindata.org/population-sources).
+    Per capita figures are calculated by dividing by a population dataset that is built and maintained by Our World in Data, based on [different sources](https://ourworldindata.org/grapher/population).
```
[the remark this answers is not in the document]
Comment (pabloarosado) [reply etl/steps/data/garden/energy/2024-06-20/energy_mix.meta.yml:43]: OK, I've written the population-sources back. But at some point we will probably stop using that and use the population data page (by next update, happening soon).

File: etl/steps/data/garden/energy_institute/2024-06-20/statistical_review_of_world_energy.meta.yml [hunk 3]
```diff
@@ -634,41 +634,41 @@ tables:
...
-        short_unit: '$'
-      coal_price_china_qinhuangdao_spot_current_dollars_per_tonne:
-        title: Coal price - China Qinhuangdao spot price
-        unit: 'current US$ per tonne'
-        short_unit: '$'
-      coal_price_japan_steam_spot_cif_current_dollars_per_tonne:
-        title: Coal price - Japan steam spot CIF price
-        unit: 'current US$ per tonne'
-        short_unit: '$'
+      # coal_price_us_central_appalachian_spot_price_index_current_dollars_per_tonne:
```
[the remark this answers is not in the document]
Comment (pabloarosado) [reply etl/steps/data/garden/energy_institute/2024-06-20/statistical_review_of_world_energy.meta.yml:640]: It's a bit annoying, but useful to see what has changed from one version to the next, in terms of these price indicators.

File: etl/steps/data/garden/energy/2024-06-20/energy_mix.meta.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (paarriagadap) [reply etl/steps/data/garden/energy/2024-06-20/energy_mix.meta.yml:43]: Ah, right. I would be happy to use the data page in this one, if it had description_key (s). That's why the article looks more complete

**Pull request merged**